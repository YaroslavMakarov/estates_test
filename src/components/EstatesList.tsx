import React from 'react';
import { useSelector } from 'react-redux';

import Estate from './Estate';

import { EstatesWrapper } from '../styled/estatesStyles';
import { estatesSelector } from '../redux/estates';

const EstatesLists = () => {
    const estates = useSelector(estatesSelector);

    return (
        <EstatesWrapper>
            {estates.map(estate => (
                <Estate
                  key={estate.id}
                  estate={estate}
                />
            ))}
        </EstatesWrapper>
    );
};

export default EstatesLists;
