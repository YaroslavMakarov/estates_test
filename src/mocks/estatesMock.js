export const estatesMock = {
    estates: {
        estates: [
            {
                'id': 1,
                'title': 'Shevchenko estate',
                'price': 30000,
                'address': {
                    'street': 'Melnychna',
                    'number': 12
                },
                'preview_img': 'https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg',
                'seller': 'Chack Norris',
                'description': 'The best from Chuck',
                'detailed_photos': [
                    'https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg',
                    'https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg',
                    'https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg'
                ]
            },
            {
                "id": 2,
                "title": "Yanukovich estate",
                "price": 3000000,
                "address": {
                    "street": "Prisident",
                    "number": 4
                },
                "preview_img": "https://biz.liga.net/images/general/2018/11/15/thumbnail-tw-20181115111659-8474.jpg?v=1542279867",
                "seller": "Chack Norris",
                "description": "The best from Chuck",
                "detailed_photos": [
                    "https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg",
                    "https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg",
                    "https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg"
                    ]
             },
        ],
    },
};