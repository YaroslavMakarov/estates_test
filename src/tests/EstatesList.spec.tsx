import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import * as ReactRedux from 'react-redux';
import configureStore from 'redux-mock-store';

import EstatesList from '../components/EstatesList';
import { estatesMock } from '../mocks/estatesMock';


const mockStore = configureStore();

describe("EstateList component", () => {
    let wrapper: ReactWrapper;
    const mockedStore = mockStore(estatesMock);
    let useSelector: jest.SpyInstance<unknown, [selector: (state: unknown) => unknown, equalityFn?: ((left: unknown, right: unknown) => boolean) | undefined]>;

    beforeEach(() => {
        useSelector = jest.spyOn(ReactRedux, 'useSelector')
            .mockImplementation(f => f(mockedStore.getState()));
          
        wrapper = mount(
            <Provider store={mockedStore}>
                <MemoryRouter>
                    <EstatesList />
                </MemoryRouter>
            </Provider>
        );
    });

    it("shoud get snapShot estates list", () => {        
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("shoud find all Estate component", () => {
        const component = wrapper.find('Estate');
        expect(component.length).toBe(1);
    });
});