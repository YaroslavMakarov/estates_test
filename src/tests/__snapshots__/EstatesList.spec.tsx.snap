// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EstateList component shoud get snapShot estates list 1`] = `
"<Provider store={{...}}>
  <MemoryRouter>
    <Router history={{...}}>
      <EstatesLists>
        <styled.div>
          <div className=\\"sc-hKgILt gYRjSP\\">
            <Estate estate={{...}}>
              <styled.div as={{...}} to=\\"/info/1\\">
                <Link to=\\"/info/1\\" className=\\"sc-bdfBwQ fhkZDT\\">
                  <LinkAnchor className=\\"sc-bdfBwQ fhkZDT\\" href=\\"/info/1\\" navigate={[Function: navigate]}>
                    <a className=\\"sc-bdfBwQ fhkZDT\\" href=\\"/info/1\\" onClick={[Function: onClick]}>
                      <styled.img src=\\"https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg\\">
                        <img src=\\"https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg\\" className=\\"sc-gsTCUz esHetz\\" />
                      </styled.img>
                      <styled.div>
                        <div className=\\"sc-dlfnbm hMEdSQ\\">
                          Name: Shevchenko estate
                        </div>
                      </styled.div>
                      <styled.div>
                        <div className=\\"sc-dlfnbm hMEdSQ\\">
                          Price: 30000$
                        </div>
                      </styled.div>
                      <styled.div>
                        <div className=\\"sc-dlfnbm hMEdSQ\\">
                          Address: Melnychna, 12
                        </div>
                      </styled.div>
                    </a>
                  </LinkAnchor>
                </Link>
              </styled.div>
            </Estate>
          </div>
        </styled.div>
      </EstatesLists>
    </Router>
  </MemoryRouter>
</Provider>"
`;
